scripts/set_config.py

```diff
@@ -113,14 +112,19 @@ def parse_args(argv):
 
     return parser.parse_args(argv)
 
-def create_config_obj(argv, log=None):
+def create_config_file(argv, config_dict=None, log_name=None):
     '''Main section for processing config file'''
 
     user_args = parse_args(argv)
 
-    if log is None:
-        name = f"{inspect.stack()[0][3]}"
-        log = cli_helpers.setup_logging(user_args, log_name=name)
+    name = log_name or f"{inspect.stack()[0][3]}"
```

Comment:
Out of curiosity, given that I haven't seen these tools running yet: Is `inspect.stack()[0][3]` the name of this function? And is that being incorporated into log messages that users see? If so, is that maybe unnecessary detail for a user-facing CLI tool, exposing internals that they shouldn't need to know about? I can see that it could be valuable for diagnostics for developers; if so, could it maybe be enabled by some sort of debug switch? There are also other tracing tools developers can use that might be even more helpful.

Reply:
As I've been refactoring on the `condev` branch, I can answer my own question: That `stack()` call indeed returns the name of the function. I think we can vastly simplify how logging is being done in `uwtools` -- it's currently complicated, with logging being set up in many places -- sometimes more than once is a given call path! -- and I'm not seeing what benefit the complexity is providing. At some point maybe we can sync up on requirements / desired behavior and I can see whether the much simpler pattern I'm used to employing would meet those.

Reply:
It is the name of the function. TIL that it's not a list that you actually get back from `inspect.stack()`, but a list of `FrameInfo` objects which are named tuples (since Python 3.5) and now class instances backward compatible with the named tuples (since Python 3.11). 

We can at least use their names here for clarity instead of their index, which is nice and self-documenting.

I am 100% on board to learn about and implement better logging tools. Using this approach was the solution to DRYing the output to include the function name.

May I propose the use of the `FrameInfo` attributes here for a clearer current DRY version and leaving the broader logging discussion for a later ticket?

Reply:
I still (lol, looking forward to the upcoming demo!) haven't seen these tools used in practice, but if we're exposing internals of our code (e.g. function names) in user output, I think that would be worth reconsidering. I know that's useful for developers, but there are other/better tools -- debuggers, tracers, profilers, etc. -- for developers to glean internals insight. I suspect that users would value clean, reliable, concise output. For example, if we did internal refactoring that changed function names, which then changed the format of log output, I don't think users necessarily need to be exposed to that. But is that what we actually have now, function names in log output that users see? And, if not, what benefit do we get from involving the function name in logging?

If there's a chance that we could revisit logging requirements and then potentially greatly simplify what we're doing, I'd say don't spend a minute changing the current code.

Reply:
I maybe missed a piece of this bigger picture before. We should indeed have this discussion (and demo!) in real life. Here, I think we are naming the logger itself so that we can make sure we retrieve the correct one (in the case where there are multiple instantiated). We may or may not actually be using it in the output information.

Reply:
> make sure we retrieve the correct one
The question I'll ask later, in a more free-form, non-PR discussion: Do we need multiple loggers? (Maybe we do! But I never have. Maybe I've just gotten lucky so far.) I might make the outrageous claim, later, that we should have a single logger, and that we should log to `stdout` and let users redirect however they like. I may well be missing the needs of some use case.

Reply:
We should probably consider a few scenarios about how the Apps have been handling loggers. We can certainly propose an alternative, if needed, and see how workflow-tools can fit in there appropriately.